File: ffmpeg/android-ffmpeg/src/main/java/nl/bravobit/ffmpeg/FFtask.java
package nl.bravobit.ffmpeg;

public interface FFtask {

    /**
     * Sends 'q' to the ff binary running process asynchronously
     */
    void sendQuitSignal();

    /**
     * Checks if process is completed
     * @return <code>true</code> if a process is running
     */
    boolean isProcessCompleted();

    /**
     * Kill given running process
     *
     * @return true if process is killed successfully
     */
    boolean killRunningProcess();
}

File: ffmpeg/android-ffmpeg/src/main/java/nl/bravobit/ffmpeg/FFtask.kt
package nl.bravobit.ffmpeg

interface FFtask {
    /**
     * Sends 'q' to the ff binary running process asynchronously
     */
    fun sendQuitSignal()

    /**
     * Checks if process is completed
     * @return `true` if a process is running
     */
    val isProcessCompleted: Boolean

    /**
     * Kill given running process
     *
     * @return true if process is killed successfully
     */
    fun killRunningProcess(): Boolean
}